### PR TITLE
Oop

### PR DESCRIPTION
**Pull request by bamsi**
In this pull request, I have implemented the following:

- Create class Person 
- Create class Student which Inherits from Person.
- Create a class Teacher who Inherits from Person.
- Each class should be saved in a separate file.

Your code review is much appreciated!

### CODE REVIEW

**Review by Mihndim2020 — COMMENTED** [review]
# STATUS: INVALID!!!  :x: :x: :x: 

Hi @bamsi,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code, please make sure your implement the following changes before you resubmit for review. 

## Required Changes ♻️
_Check the comments below._
- [ ] As it is a [requirement](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_people.md) for this project, " `Make sure that you used correct Gitflow.` ", please make sure you use [Gitflow(https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

_Invalid Code Review Request **does not count** into the code reviews limit._

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  private
+
+  def of_age?
+    (@age >= 18)
+  end
+
+  public
+
+  def can_use_services?
+    is_of_age? || @parent_permission
```
Comment (ShadyShawkat) [inline person.rb]: - [ ] Please make sure that you're calling the right method here. You've named the method `of_age?` but when you called you, you typed `is_of_age?`

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi @bamsi, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DammyShittu — APPROVED** [review]
Hi @bamsi,

## APPROVED 🟢

## Highlights
✔️ Use of private methods

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**